### PR TITLE
Create abuse_docusign_unsolicited_reply-to.yml

### PR DESCRIPTION
**Pull request by zoomequipd**
# Description

Create new rule for DocuSign shares which contain a reply-to address or domain that has not been in previous contact with the recipient organization

# Associated samples

- [Sample 1](https://platform.sublime.security/messages/66a153d700cff7d1424887b51802b2a310ea57fd32eb55b1407c57c8b79412ed)

### CODE REVIEW

File: detection-rules/abuse_docusign_unsolicited_reply-to.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    )
+  )
+  and headers.auth_summary.dmarc.pass
+  
+  // not a completed DocuSign
+  // reminders are sent automatically and can be just as malicious as the initial
+  // users often decline malicious ones
+  and not strings.istarts_with(subject.subject, "Completed:")
+  
+  // reply-to email address as never been sent an email by the org
```
Comment (aidenmitchell) [inline detection-rules/abuse_docusign_unsolicited_reply-to.yml:25]:
```suggestion
  // reply-to email address has never been sent an email by the org
```

File: detection-rules/abuse_docusign_unsolicited_reply-to.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    any(headers.reply_to, .email.email in $recipient_emails)
+    // if the reply-to email address is NOT in free_email_providers, check the domain in recipient_domains
+    or any(filter(headers.reply_to,
+                  // filter the list to only emails that are not in free_email_providers
+                  (
+                    .email.domain.domain not in $free_email_providers
+                    or .email.domain.root_domain not in $free_email_providers
+                  )
+           ),
+           .email.domain.root_domain in $recipient_domains
```
Comment (aidenmitchell) [inline detection-rules/abuse_docusign_unsolicited_reply-to.yml:36]: You might want `.email.domain.domain`, IIRC the list is of domains and not root domains

File: detection-rules/abuse_docusign_unsolicited_reply-to.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    any(headers.reply_to, .email.email in $sender_emails)
+    // if the reply-to address is NOT in free_email_providers, check the domain in sender_domains
+    or any(filter(headers.reply_to,
+                  // filter the list to only emails that are not in free_email_providers
+                  (
+                    .email.domain.domain not in $free_email_providers
+                    or .email.domain.root_domain not in $free_email_providers
+                  )
+           ),
+           .email.domain.root_domain in $sender_domains
```
Comment (aidenmitchell) [inline detection-rules/abuse_docusign_unsolicited_reply-to.yml:50]: You might want `.email.domain.domain`, IIRC the list is of domains and not root domains

**Comment by zoomequipd** [comment]
pulled in https://github.com/sublime-security/sublime-rules/pull/2090

**Review by morriscode — APPROVED** [review]

**Pull request merged**